includes/class-webmention-receiver.php

```diff
@@ -50,6 +53,40 @@ public static function init() {
 		self::register_meta();
 	}
 
+	public static function comment_row_actions( $actions, $comment ) {
+		$approve_nonce = esc_html( '_wpnonce=' . wp_create_nonce( "approve-comment_$comment->comment_ID" ) );
+		$approve_url   = esc_url( "comment.php?action=approvecomment&$approve_nonce" );
+		$unapprove_url = esc_url( "comment.php?action=unapprovecomment&$approve_nonce" );
+		$status        = wp_get_comment_status( $comment );
+		if ( 'unapproved' === $status ) {
+			$actions['domainwhitelist'] = "<a href='{$approve_url}&domain=true&c={$comment->comment_ID}' aria-label='" . esc_attr__( 'Approve & Whitelist', 'webmention' ) . "'>" . __( 'Approve & Whitelist', 'webmention' ) . '</a>';
+		}
+		return $actions;
+	}
+
+	public static function transition_to_whitelist( $comment ) {
+		if ( ! current_user_can( 'moderate_comments' ) ) {
+			return;
+		}
+		if ( isset( $_REQUEST['domain'] ) ) {
+			$url = get_comment_meta( $comment->comment_ID, 'webmention_source_url', true );
+			if ( ! $url ) {
+				return;
+			}
+			$whitelist = get_option( 'webmention_approve_domains' );
```

Comment:
can you encapsulate this in some functions? For example `get_webmention_approve_domains` and `add_webmention_approve_domains`... (just example names, feel free to change them to a better/more readable version)

Reply:
Will do.